// @daypilot/daypilot-lite-angular/esm2020/lib/daypilot.module.mjs
import { NgModule } from '@angular/core';
import { DayPilotCalendarComponent } from './daypilot-calendar.component';
import { DayPilotMonthComponent } from "./daypilot-month.component";
import { DayPilotNavigatorComponent } from "./daypilot-navigator.component";
import * as i0 from "@angular/core";
const COMPONENTS = [
    DayPilotCalendarComponent,
    DayPilotMonthComponent,
    DayPilotNavigatorComponent
];
export class DayPilotModule {
}
DayPilotModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: DayPilotModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DayPilotModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: DayPilotModule, declarations: [DayPilotCalendarComponent,
        DayPilotMonthComponent,
        DayPilotNavigatorComponent], exports: [DayPilotCalendarComponent,
        DayPilotMonthComponent,
        DayPilotNavigatorComponent] });
DayPilotModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: DayPilotModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.0", ngImport: i0, type: DayPilotModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ...COMPONENTS
                    ],
                    imports: [],
                    exports: [
                        ...COMPONENTS
                    ]
                }]
        }] });
